src/audio/module_adapter/module_adapter.c

```diff
@@ -123,6 +123,19 @@ struct comp_dev *module_adapter_new(const struct comp_driver *drv,
 
 	dev->state = COMP_STATE_READY;
 
+#if CONFIG_IPC_MAJOR_4
+	struct sof_ipc_stream_params params = {{ 0 }};
+
+	/* set the stream params based on the module base cfg. It is OK to pass NULL params
+	 * here because it will be filled in based on the module base_cfg.
+	 */
+	ret = module_adapter_params(dev, &params);
```

Comment:
in fact clearing `params` here isn't even needed - `module_adapter_params()` -> `module_adapter_set_params()` -> `ipc4_base_module_cfg_to_stream_params()` -> `memset(0)`